src/components/trie/utils.ts

```diff
@@ -22,46 +33,26 @@ export function trieToTreeUI(
     const branch = root.asBranchNode();
     const leftHash = branch.getLeft();
     const rightHash = branch.getRight();
-    const left = trieToTreeUI(nodes.get(leftHash), nodes, hideEmpty);
-    const right = trieToTreeUI(nodes.get(rightHash), nodes, hideEmpty);
-    const children = [];
 
-    if (shouldRenderNode(leftHash, hideEmpty)) {
-      children.push({
-        name: leftHash.toString(),
-        children:
-          Array.isArray(left) || left === undefined
-            ? left
-            : nodes.get(leftHash)?.getNodeType() === NodeType.Branch
-              ? left.children
-              : [left],
-      });
-    }
-    if (shouldRenderNode(rightHash, hideEmpty)) {
-      children.push({
-        name: rightHash.toString(),
-        children:
-          Array.isArray(right) || right === undefined
-            ? right
-            : nodes.get(rightHash)?.getNodeType() === NodeType.Branch
-              ? right.children
-              : [right],
-      });
-    }
+    const left = trieToTreeUI(nodes.get(leftHash), leftHash, nodes, hideEmpty);
+    const right = trieToTreeUI(nodes.get(rightHash), rightHash, nodes, hideEmpty);
 
     return {
-      name: "Root",
-      children,
+      name: hash.toString(),
+      children: [
+        shouldRenderNode(leftHash, hideEmpty) ? left : null,
+        shouldRenderNode(rightHash, hideEmpty) ? right : null,
+      ].filter((x) => x) as TreeNode[],
     };
   }
 
   const leaf = root.asLeafNode();
   const valueLength = leaf.getValueLength();
 
-  // return { name: `Leaf('${leaf.getKey().toString()}',${value})` };
   return {
     name: leaf.getKey().toString(),
     attributes: {
+      parentHash: hash.toString(),
```

Comment:
```suggestion
      key: leaf.getKey().toString()
```

This is not parent hash, the `hash` contains the actual hash of the leaf. But the leaf also has a `key` that we should display there.

Reply:
I made it on purpose. I had a conversation with Piotr about correlation between the user input and values in the tree. I swapped hash with value key to make it more noticeable for the user where in the tree is the exact element (key) provided on the left side. Although, I agree it doesn't feel right from the algorithm perspective cause it's not an actual hash to the element, but it has the same format.

Reply:
It's fine - you can display the key directly on the node, but please fix the tooltip then, because it now incorrectly says:
```
Hash: <key>
ParentHash: <hash> (no such thing as parent hash)
```